slack_notification/slack_notification.php

```diff
@@ -84,32 +89,55 @@
     // Build an array of fields to be rendered with Slack Attachments as a table
     // attachment-style formatting:
     // https://api.slack.com/docs/attachments
-    $fields += array(
-      array(
-        'title' => 'Commit',
-        'value' => rtrim($hash),
-        'short' => 'true'
-      ),
-      array(
-        'title' => 'Commit Message',
-        'value' => $message,
-        'short' => 'false'
-      )
+    $fields[] = array(
+      'title' => 'Commit',
+      'value' => rtrim($hash),
+      'short' => 'true'
+    );
+    $fields[] = array(
+      'title' => 'Commit Message',
+      'value' => $message,
+      'short' => 'false'
     );
+    $attachment += array('pretext' => 'Code syncing');
+    break;
+
+  case 'create_cloud_development_environment':
+    // Prepare the slack payload as per:
+    // https://api.slack.com/incoming-webhooks
+    $text = $_ENV['PANTHEON_ENVIRONMENT'] . ' spinning up for';
+    $text .= $_ENV['PANTHEON_SITE_NAME'] .' by '. $_POST['user_email'] .' complete!';
+    $text .= ' <https://dashboard.pantheon.io/sites/'. PANTHEON_SITE .'#'. PANTHEON_ENVIRONMENT .'/deploys|View Dashboard>';
+    $attachment += array('pretext' => 'New environment created');
+    break;
+
+  case 'cache_clear':
+    // Prepare the slack payload as per:
+    // https://api.slack.com/incoming-webhooks
+    $text = 'Cache clear for '. $_ENV['PANTHEON_ENVIRONMENT'] . ' on ';
+    $text .= $_ENV['PANTHEON_SITE_NAME'] .' by '. $_POST['user_email'] .' complete!';
+    $text .= ' <https://dashboard.pantheon.io/sites/'. PANTHEON_SITE .'#'. PANTHEON_ENVIRONMENT .'/deploys|View Dashboard>';
+    $attachment += array('pretext' => 'Caches cleared');
     break;
 
   default:
     $text = $_POST['qs_description'];
     break;
 }
 
-$attachment = array(
+// Add the prepared $text and $fields arrays
+$attachment += array(
   'fallback' => $text,
-  'pretext' => 'Deploying :rocket:',
-  'color' => $pantheon_yellow, // Can either be one of 'good', 'warning', 'danger', or any hex color code
   'fields' => $fields
 );
 
+// Add pretext default if not already defined
+if (empty($attachment->pretext)) {
+  $attachment += array(
+    'pretext' => 'Pantheon system update :rocket:'
+  );
+}
+
 _slack_notification($secrets['slack_url'], $secrets['slack_channel'], $secrets['slack_username'], $text, $attachment, $secrets['always_show_text']);
```

Comment:
I was looking at:

https://github.com/jordan8037310/quicksilver-examples/commit/c475bd192158c824319d51e10972f743fb951449

We removed the channel from the code intentionally because if left blank the default channel the webhook is configured to post into should be used, which is what developers were expecting. Unless something's changed, you don't need to specify a `channel` in order for it to work.

https://github.com/pantheon-systems/quicksilver-examples/commit/27a3251eb68462e484b30a240dca00ba2f23a92c

However, we clearly missed taking it end-to-end, hence the confusion.